django_auth_lti/patch_reverse.py

```diff
@@ -0,0 +1,33 @@
+"""
+Monkey-patch django.core.urlresolvers.reverse to add resource_link_id to all URLs
+"""
+from django.core import urlresolvers
+
+from .thread_local import get_current_request
+
+
+django_reverse = None
+
+
+def reverse(*args, **kwargs):
+    """
+    Call django's reverse function and append the current resource_link_id as a query parameter
+    """
+    request = get_current_request()
+    url = django_reverse(*args, **kwargs)
+    if '?' not in url:
+        url += '?'
+    return "%s&resource_link_id=%s" % (url, request.LTI.get('resource_link_id'))
```

Comment:
Can we use [urlparse](https://docs.python.org/2.7/library/urlparse.html) to do this, instead of by hand?